### PR TITLE
Op/RGW Expose DNS configuration to CRD

### PR DESCRIPTION
**Pull request by dgonzalezruiz**
This makes DNS settings configurable, so that object stores can be accessed from different hostnames than the service name.

### CODE REVIEW

**Comment by dgonzalezruiz** [comment]
It seems that I forgot to implement the Rook API possibility (and test it).

File: cmd/rook/rgw.go [hunk 1]
```diff
@@ -42,7 +43,8 @@ var (
 func init() {
 	rgwCmd.Flags().StringVar(&rgwName, "rgw-name", "", "name of the object store")
 	rgwCmd.Flags().StringVar(&rgwKeyring, "rgw-keyring", "", "the rgw keyring")
-	rgwCmd.Flags().StringVar(&rgwHost, "rgw-host", "", "dns host name")
+	rgwCmd.Flags().StringVar(&rgwHost, "rgw-host", "", "RGW hostname")
```
Comment (kokhang) [inline cmd/rook/rgw.go]: Can you tell me what is the difference between the two config? rgw-host vs rgw-dns-name?. Do we still need the formal?

Comment (dgonzalezruiz) [reply cmd/rook/rgw.go]: The hostname is still set in the rgw properties, so both are still required. Thanks for the feedback!!

**Comment by dgonzalezruiz** [comment]
Also, something worth mentioning, is that all the integration testing we do over objectstores is IP related. Given that there are no dns addresses being used anywhere, this issue has never surfaced.

**Comment by dgonzalezruiz** [comment]
I haven't been able to find a graceful way of adding Host headers with the go sdk, so, since after some experiments, the DNS issues are reproducible with simple http requests, I will try that way.

File: pkg/operator/rgw/rgw.go [hunk 2]
```diff
@@ -394,6 +397,11 @@ func (s *ObjectStore) rgwContainer(version string) v1.Container {
 		container.Args = append(container.Args, fmt.Sprintf("--rgw-cert=%s", path))
 	}
 
+	if s.Spec.Gateway.DnsName == "" {
+		container.Args[4] = fmt.Sprintf("--rgw-dns-name=%s", s.instanceName())
```
Comment (dgonzalezruiz) [inline pkg/operator/rgw/rgw.go]: As noted by some slack users, this should be the instanceName + the namespace

**Comment by galexrt** [comment]
I'm guessing now but is there a way to specify multiple dns names/hosts? If so, I think we should cover "every" possible way inside of Kubernetes already.
For example an object store with the service name `myrgw` in the namespace `mynamespace` should have all these as a host and dns name:
* `myrgw`
* `myrgw.mynamespace`
* `myrgw.mynamespace.svc`
* `myrgw.mynamespace.svc.cluster.local` (and when it is possible to get the cluster DNS prefix)

**Comment by galexrt** [comment]
@dgonzalezruiz I would be for turning it into an array of strings, as said if it is possible we should a) set all names the service may have in Kubernetes and b) allow the user to specify "as many as the user wants".

**Comment by jbguerraz** [comment]
May I ask why did it get closed ?

**Pull request closed without merging**